labs/b4.md

```diff
@@ -0,0 +1,269 @@
+(libre) software experience. In the context of free software, "libre" means
+that the source code is available under a permissive license, enabling
+downstream modification and rediistribution. Many other distributions are based
+on Debian, as Debian releases are usually very stable, meaning that the
+software provided by the distribution has been tested to be as bug-free as
+possible. One such distribution is Ubuntu, which is very popular and has
+spawned its own family of distributions. What defines the Debian family of
+distributions is the use of the DEB package format and the dpkg package
+manager.
+
+## But what about these “package managers”?
+
+If you come from the Windows or MacOS ecosystems, you might be wondering what a
+package manager is. On those operating systems, the most common method of
+installing new programs is to run an installer executable that unpacks and
+copies files into the correct location, and updates the OS registry to reflect
+this.
+
+Enter the package manager, a centralized way to install, update, and remove
+software from your computer using verified sources called repositories.  Most
+package managers work in the same way: the package manager gets a list of
+packages from the repository, and then, when asked to install a package,
+fetches the package from the repository, verifies that it is legit, and
+installs it. A package manager is like a librarian. When a patron wants to read
+a book, the librarian consults a catalog (which is the package database) and
+then fetches the book from the shelf and gives it to the patron (installing the
+package).  Sometimes, the librarian has to update the catalog because new
+editions of the books were added (updating the package database). When the
+patron is done, the librarian knows where to return the book (cleaning up after
+removing a package).  Contrast this to the approach on Windows, where dedicated
+uninstaller programs are occasionally necessary as the OS does not know where
+programs install their own files.
+
+## Debian: An Example
+
+In this class, we will be focused on using Debian. As noted before, Debian uses
+`apt` or `dpkg` as its package manager.
+
+We invoke the package manager by using `apt`. Before installing anything, you
+shoud update your local copy of packages and versions to ensure the package
+manager installs the latest and most up-to-date packages. To do that, run:
+
+    $ sudo apt update
+
+To find the package to install:
+
+    $ apt search [package|description]
+
+To install a package, run:
+
+    $ sudo apt install package
+
+To remove a package, run:
+
+    $ sudo apt remove package
+
+Easy? When you want to upgrade the packages that you have installed when
+new versions are released, you can do so by calling:
+
+    $ sudo apt upgrade
+
+There are also other commands, such as removing unneeded dependencies (`sudo
+apt autoremove`) and purging packages (`sudo apt purge`) but that is what the
```

Comment:
Explain the difference between purging and removing

Reply:
We reference them to the man pages, I think that is ok.